compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathChangesComputer.kt

```diff
@@ -246,6 +246,22 @@ object ClasspathChangesComputer {
         //     classes, and symbols in removed classes.
         incrementalJvmCache.clearCacheForRemovedClasses(changesCollector)
 
+        // IncrementalJvmCache currently doesn't use the `KotlinClassInfo.extraInfo.classSnapshotExcludingMembers` info when comparing
+        // classes, so we need to do it here.
+        // TODO(KT-58289): Ensure IncrementalJvmCache uses that info when comparing classes.
+        val currentClassSnapshotsExcludingMembers = currentClassSnapshots
+            .associate { it.classId to it.classMemberLevelSnapshot!!.extraInfo.classSnapshotExcludingMembers }
+            .filter { it.value != null }
+        val previousClassSnapshotsExcludingMembers = previousClassSnapshots
+            .associate { it.classId to it.classMemberLevelSnapshot!!.extraInfo.classSnapshotExcludingMembers }
+            .filter { it.value != null }
```

Comment:
We can optimize this a bit more:
````
val currentClassSnapshotsExcludingMembers = currentClassSnapshots.assocaite { key to value }
previousClassSnapshots.forEach {
  if (it.cassId in currentClassSnapshotsExcludingMembers) {
    // compare them
  }
}
```

Reply:
Yup, done in the next PR ([https://github.com/JetBrains/kotlin/pull/5147](https://github.com/JetBrains/kotlin/pull/5147)).